library/std/src/path.rs

```diff
@@ -334,6 +334,29 @@ fn split_file_at_dot(file: &OsStr) -> (Option<&OsStr>, Option<&OsStr>) {
     }
 }
 
+fn split_file_at_dot(file: &OsStr) -> (Option<&OsStr>, Option<&OsStr>) {
+    let slice = os_str_as_u8_slice(file);
+    if slice == b".." {
+        return (Some(file), None);
+    }
+
+    // The unsafety here stems from converting between &OsStr and &[u8]
+    // and back. This is safe to do because (1) we only look at ASCII
+    // contents of the encoding and (2) new &OsStr values are produced
+    // only from ASCII-bounded slices of existing &OsStr values.
+    let i = match slice[1..].iter().position(|b| *b == b'.') {
```

Comment:
This doesn't match the documented behavior on `Path::new("..x").file_prefix()`. The documentation suggests it would return `""` but it actually returns `"."`.

> * `None`, if there is no file name;

does not apply

> * The entire file name if there is no embedded `.`;

does not apply

> * The entire file name if the file name begins with `.` and has no other `.`s within;

does not apply

> * Otherwise, the portion of the file name before the first `.`

The portion of the file name before the first `.` is empty string.

Reply:
You are correct, this is misleading. My intended functionality - to be explicit - was "the portion of the file name before the first `.` (excluding where the first character if the file name is `.`)"

As an example, I would expect/like `.hidden.file` to return `.hidden` as the `file_prefix`.

Should I 
1. Update the documentation to make this explicit (my original intention)
2. Change the implementation so that `.hidden.file` returns `None`?

Reply:
I've also added a test in a889529e984f1d477153e60c16fa4066d8b6c9d7 that makes the expected behaviour very explicit for this.

Reply:
I would recommend updating the documentation so that it accurately describes the implemented behavior. It's fine to make the implementation do what your intended functionality is, but the documentation needs to not conflict with that.